### ISSUE TITLE
Remove GeoPandas dependency

### ISSUE THREAD

**Issue by mdtanker** [mdtanker/polartoolkit#18]
**Description of the desired feature:**

GeoPandas is causing some issues with Poetry and packaging/install on a Windows machine, I think mostly due to its C dependencies. Since we only use it to read shapefiles, there might be an alternative.

* [This thread](https://github.com/geopandas/geopandas/issues/1313#issuecomment-594067848) suggests `Pandas/Shapely` or `Spatialpandas`.

* Or maybe [pyshp](https://github.com/GeospatialPython/pyshp) with this [Gist](https://gist.github.com/aerispaha/f098916ac041c286ae92d037ba5c37ba)

* Look here for discussion on [spatialpandas](https://github.com/holoviz/spatialpandas/issues/73)

* [Article](https://geospatialtraining.com/tutorial-creating-a-pandas-dataframe-from-a-shapefile/) on using [Arcpy](https://pro.arcgis.com/en/pro-app/2.8/arcpy/get-started/installing-arcpy.htm)

* [Link to using spark.read.shapefile()](https://docs.astraea.earth/hc/en-us/articles/360043904011-Read-a-Shapefile-into-a-GeoPandas-DataFrame) 

* [Another similar one](https://docs.astraea.earth/hc/en-us/articles/360043452452-Read-a-Shapefile-into-Spark-DataFrame)

**Are you willing to help implement and maintain this feature?**

**Comment by mdtanker** [comment]
Two more thinks to consider:

* look into using [dask-geopandas](https://dask-geopandas.readthedocs.io/en/latest/) to speed up reading large shapefiles.

* maybe switch to [geopandas-base](https://github.com/conda-forge/geopandas-feedstock) to reduce dependencies.

**Comment by weiji14** [comment]
Have you checked out https://github.com/geopandas/pyogrio?

**Comment by mdtanker** [comment]
Look's like it still requires GeoPandas to read a .shp  😑, so I think I'd still need to include GeoPandas in the .toml dependencies, which won't work on Windows, unless I'm missing something. Thanks though  :smile:

**Comment by weiji14** [comment]
Ah yes, that's true. But I think `pyogrio` also ships with GDAL in the wheels (see https://github.com/geopandas/pyogrio/issues/48, and notice at https://pypi.org/project/pyogrio/0.4.1/#files how the .whl files are quite big) whereas `geopandas` doesn't ship with GDAL. So if you do `pip install pyogrio[geopandas]`, you should be indirectly getting GDAL already, if that was the original issue on Windows.

For reference, if you're using poetry, the `pyogrio[geopandas]` dependency line would look like `pyogrio = {version = ">=0.4.0", extras = ["geopandas"]}` (e.g. at https://github.com/weiji14/zen3geo/blob/71e886d95454de70651cc31ea6dedc33e929145c/pyproject.toml#L25)

**Comment by mdtanker** [comment]
Amazing! Thanks for that 👍 I originally tried `pip install pyogrio[geopandas]` which wasn't working, but `pip install --pre pyogrio[geopandas]` did the trick. I'll add that in my .toml to save some hassle for separately install geopandas with conda.

Any idea if a similar approach will be taken with`PyGMT` to include `GMT` in a pip install?

**Comment by weiji14** [comment]
> Amazing! Thanks for that 👍 I originally tried `pip install pyogrio[geopandas]` which wasn't working, but `pip install --pre pyogrio[geopandas]` did the trick. I'll add that in my .toml to save some hassle for separately install geopandas with conda.

Cool, glad that it worked!

> Any idea if a similar approach will be taken with`PyGMT` to include `GMT` in a pip install? 

Some day perhaps, see https://github.com/GenericMappingTools/pygmt/issues/1853 :wink:

**Comment by mdtanker** [comment]
I've tried to replace `geopandas` with `pyogrio` in poetry, but am still having issues. It worked fine to just add `pyogrio`, but to read a shapefile, `geopandas` still needs to be installed. This can be done with `pip install geopandas --no-deps`, but I haven't found a way to get poetry to add `geopandas` without also adding its dependency `fiona`, which causes issues on windows.

For now I've added `pyogrio` to the dependencies, but will inform users to separately install `geopandas`.
